src/wallet/walletdb.cpp

```diff
@@ -452,7 +447,13 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
         }
         else if (strType == "defaultkey")
         {
-            ssValue >> pwallet->vchDefaultKey;
+            CPubKey vchPubKey;
+            ssKey >> vchPubKey;
+            if (!vchPubKey.IsValid())
+            {
+                strErr = "Error reading wallet database: Default Key corrupt";
```

Comment:
"defaultkey" is not part of `IsKeyType`, so this failure will just be a non-critical warning upon loading. I assume that's intentional?